### ISSUE TITLE
Monit does not monitor Kubelet/Kube-proxy on master

### ISSUE THREAD

**Issue by zmerlynn** [GoogleCloudPlatform/kubernetes#8171]
Noticed on internal tight cluster-up/down loop, one of our clusters failed to come up. I logged into the failed cluster, and none of the master components were running. (`docker ps` is empty). Checked docker logs, and it's clear we started them all .. and then knocked Docker over. Looked at monit logs, and yup:

```
$ less /var/log/monit.log 
[UTC May 13 06:22:40] info     : monit: generated unique Monit id 4cc35451ed20ffd0da71aa05d57c35e5 and stored to '/var/lib/monit/id'
[UTC May 13 06:22:40] info     : Starting monit daemon
[UTC May 13 06:22:40] error    : State file '/var/lib/monit/state': Unable to read magic
[UTC May 13 06:22:40] info     : 'system_k8s-staging-regtest-loop-2015-05-12-23-21-35-master.c.cloud-kubernetes-dev.internal' Monit started
[UTC May 13 06:22:55] info     : monit daemon with pid [2737] killed
[UTC May 13 06:22:55] info     : 'system_k8s-staging-regtest-loop-2015-05-12-23-21-35-master.c.cloud-kubernetes-dev.internal' Monit stopped
[UTC May 13 06:22:55] info     : Starting monit daemon
[UTC May 13 06:22:55] info     : 'system_k8s-staging-regtest-loop-2015-05-12-23-21-35-master.c.cloud-kubernetes-dev.internal' Monit started
[UTC May 13 06:24:55] error    : 'docker' process is not running
[UTC May 13 06:24:55] info     : 'docker' trying to restart
[UTC May 13 06:24:55] info     : 'docker' start: /etc/init.d/docker
[UTC May 13 06:26:55] info     : 'docker' process is running with pid 4503
```

What's supposed to restart the master components? Clearly kubelet doesn't seem to be doing its job here.

I still have the cluster if you want it for debugging.

cc @dchen1107 @roberthbailey

**Comment by zmerlynn** [comment]
cc @ArtfulCoder as well, of course

**Comment by zmerlynn** [comment]
Hrm, kubelet thinks it recovered the containers:
```
E0513 06:23:02.942692    3545 manager.go:148] Docker container factory registration failed: unable to communicate with docker daemon: EOF.
I0513 06:23:02.943366    3545 factory.go:64] Registering Raw factory
I0513 06:23:02.943388    3545 manager.go:897] Started watching for new ooms in manager
I0513 06:23:02.943406    3545 oomparser.go:199] OOM parser using kernel log file: "/var/log/kern.log"
I0513 06:23:02.943707    3545 manager.go:219] Starting recovery of all containers
I0513 06:23:02.973893    3545 manager.go:224] Recovery completed
```

**Comment by vmarmol** [comment]
@zmerlynn that line is from cAdvisor, it shouldn't affect Kubelet's work. I'd be interested to take a look at the cluster and Kubelet's log. I wonder if we're unable to make a connection to Docker (we never retry).

**Comment by dchen1107** [comment]
@zmerlynn From what you posted above, looks like monit restarted docker properly, and kubelet is running normally. Since I don't have kubelet log, I just randomly guessed several possibilities:

1) Some thing failed to generate proper master components manifest. Can you check /etc/kubernetes/manifests?
2) what is the output of 'ps -ef | grep kubelet'?

**Comment by zmerlynn** [comment]
I gave @vmarmol the access info.

**Comment by zmerlynn** [comment]
... but you already have access to our project, too, if you want to login. Forwarding you the email.

**Comment by vmarmol** [comment]
It appears the Kubelet is not running on the machine. Investigating...

**Comment by vmarmol** [comment]
Looks like we don't register the Kubelet (and Kube-proxy) with monit on the master. Oops :-\ the Kubelet went down 10h ago and nothing restarted it.

Monit on the master is tracking (verified on a clean cluster I have):
- Docker

Monit on the nodes is tracking:
- Kubelet
- Kube-proxy
- Docker

Looks like we only set up monit for nodes: 

https://github.com/GoogleCloudPlatform/kubernetes/blob/master/cluster/saltbase/salt/monit/init.sls#L15

@dchen1107 is working on a PR

**Comment by dchen1107** [comment]
On master node, monit only watch docker, not kubelet and kube-proxy. I will send fix soon.

**Comment by zmerlynn** [comment]
Thanks! FWIW, we're pushing 0.17.0 anyways, so unless there's a dot release soon (which I may do for a different change), it's not an emergency.

**Comment by zmerlynn** [comment]
Fixed by #8189
